lib/helpers.ts

```diff
@@ -0,0 +1,48 @@
+import { ENTRIES_PER_PAGE, LANGUAGES } from '@lib/constants';
+import _ from 'lodash';
+
+export const getNumberedStaticPaths = async (
+	sectionSegments: string,
+	getCount: (language: string) => Promise<number>
+) => {
+	const pathSetPromises = Object.keys(LANGUAGES).map(async (k) => {
+			const sermonCount = await getCount(k),
+				pageCount = Math.ceil(sermonCount / ENTRIES_PER_PAGE),
+				numbers = Array.from(Array(pageCount).keys()),
+				base = LANGUAGES[k].base_url;
+
+			return numbers.map((x) => `/${base}/${sectionSegments}/${x + 1}`);
+		}),
+		pathSets = await Promise.all(pathSetPromises);
+
+	return {
+		paths: pathSets.flat(),
+		fallback: 'unstable_blocking',
+	};
+};
+
+export async function getPaginatedStaticProps(language: string, pageIndex: number, getter: any) {
```

Comment:
There might be a more specific type but you could start with:

```suggestion
export async function getPaginatedStaticProps(language: string, pageIndex: number, getter: Promise<{ nodes: any[], aggregate: { count: number; } }>) {
```